extensions/wikia/AdEngine/js/template/bfaa.js

```diff
@@ -9,14 +9,41 @@ define('ext.wikia.adEngine.template.bfaa', [
 ], function (adContext, adHelper, doc, log, win, mercuryListener) {
 	'use strict';
 
-	var logGroup = 'ext.wikia.adEngine.template.bfaa',
+	var bfabSlotName = 'BOTTOM_LEADERBOARD',
 		breakPointWidthNotSupported = 767, // SCSS property: $breakpoint-width-not-supported
 		desktopHandler,
+		disableBtf = win.ads.runtime.disableBtf,
 		mobileHandler,
+		logGroup = 'ext.wikia.adEngine.template.bfaa',
 		nav,
 		page,
+		threshold = 100,
+		viewPortHeight = Math.max(doc.documentElement.clientHeight, win.innerHeight || 0),
+		mainContent = doc.getElementById('WikiaMainContent'),
 		wrapper;
 
+	function getTopOffset(el) {
+		var topPos = 0;
+		for (; el !== null; el = el.offsetParent) {
+			topPos += el.offsetTop;
+		}
+
+		return topPos;
+	}
+
+	function pushBfab() {
+		var scrollPosition = win.scrollY || win.pageYOffset || doc.documentElement.scrollTop,
+			pushBfabPosition = getTopOffset(mainContent) + mainContent.offsetHeight - viewPortHeight - threshold;
+
+		if (pushBfabPosition < scrollPosition) {
+			win.ads.runtime.disableBtf = false;
+			win.adslots2.push(bfabSlotName);
+			win.ads.runtime.disableBtf = disableBtf;
```

Comment:
I'm not sure of this way of "unblocking" BTF slot. Isn't it risky for race conditions?